internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -584,6 +606,25 @@ func resourcePostgresqlFlexibleServerUpdate(d *pluginsdk.ResourceData, meta inte
 		}
 	}
 
+	if d.HasChange("replication_role") {
```

Comment:
Would it make sense to omit `replication_role` entirely given that it cannot be set in create and can only be updated to `None`? Or if there a use-case for why it would be necessary to update the `replication_role` to `None`?

Reply:
When the "create_mode" is "Replica" and the "replication_role" isn't set, the server is read replica server.
When the "create_mode" is "Replica" and the "replication_role" is set to "None", the server would be prompted to read/write server.

This is a typical use-case. So I assume it's necessary. Here is the [similar change](https://github.com/hashicorp/terraform-provider-azurerm/blob/a39612ce1809a7df1d8ed4da9fdf41dc9047d0d8/internal/services/mysql/mysql_flexible_server_resource.go#L569) for replication_role in mysql fs.

Reply:
Thanks @neil-yechenwei sounds like it is needed then!